tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,21 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// Default number of lines to buffer before dropping logs if `NonBlocking` is is lossy.
 pub const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
 
+/// `WorkerGuard` is responsible for informing the writer to flush logs in the case of a panic.
+///
+/// This guard should be held for as long as possible and should not be dropped accidentally.
+/// It contains a reference to an `AtomicBool` which is used to inform the off-thread writer's
+/// worker to shutdown and flush logs.
```

Comment:
A couple notes:
- In general, idiomatic RustDoc comments don't begin with the name of the type or function they document. 
- Again, talking about the `AtomicBool` seems like an implementation detail. Instead of describing _how_ this works, I would talk about _why_ users will want to use it.

How about something like: 
```suggestion
/// A guard which triggers an associated [`NonBlocking`] writer to flush logs when dropped.
///
/// Writing to a [`NonBlocking`] writer will **not** immediately write a log line to the underlying
/// output. Instead, the log line will be enqueued to be written by the logging worker thread. In
/// addition, to improve throughput, the non-blocking writer flushes the underlying output
/// periodically, rather than every time a line is written. This means that if the program 
/// terminates abruptly (such as by panicking, or by calling `std::process::exit`), some log lines 
/// may not be written.
///
/// Since logs recorded near a crash are often necessary for diagnosing the failure, this type 
/// provides a mechanism to ensure that all buffered logs are written to the output, and that the
/// output is flushed prior to terminating. In order for this to work, this guard should generally 
/// be held in the `main` function (or whatever the outermost scope of the program is). This will 
/// ensure that it is dropped when unwinding, or when `main` returns.
```